src/controlplane/client/connector.go

```diff
@@ -98,133 +98,3 @@ func (dp *defaultConnector) probeEndpoint(url string, client *http.Client) error
-	if endpoint.Auth == nil {
-		dp.logger.Debugf("No authentication configured for %q, connection will be attempted anonymously", endpoint.URL)
-
-		return nil
-	}
-
-	if strings.EqualFold(endpoint.Auth.Type, bearerAuth) {
-		dp.logger.Debugf("Using kubernetes token to authenticate request to %q", endpoint.URL)
-
-		httpClient.Transport = transport.NewBearerAuthRoundTripper(dp.inClusterConfig.BearerToken, httpClient.Transport)
-
-		return nil
-	}
-
-	if strings.EqualFold(endpoint.Auth.Type, mTLSAuth) {
-		dp.logger.Debugf("Using mTLS to authenticate request to %q", endpoint.URL)
-
-		tlsConfig, err := dp.getTLSConfigFromSecret(endpoint.Auth.MTLS)
-		if err != nil {
-			return fmt.Errorf("could not load TLS configuration: %w", err)
-		}
-
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: tlsConfig,
-		}
-
-		return nil
-	}
-
-	return fmt.Errorf("unknown authorization type %q", endpoint.Auth.Type)
-}
-
-func (dp *defaultConnector) getTLSConfigFromSecret(mTLSConfig *config.MTLS) (*tls.Config, error) {
-	if mTLSConfig == nil {
-		return nil, fmt.Errorf("mTLS config cannot be nil")
-	}
-
-	if mTLSConfig.TLSSecretName == "" {
-		return nil, fmt.Errorf("mTLS secret name cannot be empty")
-	}
-
-	namespace := mTLSConfig.TLSSecretNamespace
-	secretName := mTLSConfig.TLSSecretName
-	if namespace == "" {
-		dp.logger.Debugf("TLS Secret name configured, but not TLS Secret namespace. Defaulting to `default` namespace.")
-		namespace = "default"
-	}
-
-	secret, err := dp.kc.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
-	if err != nil {
-		return nil, fmt.Errorf("could not find secret %q containing TLS configuration: %w", secretName, err)
-	}
-
-	var cert, key, cacert []byte
-
-	var ok bool
-	if cert, ok = secret.Data["cert"]; !ok {
-		return nil, fmt.Errorf("could not find TLS certificate in `cert` field in secret %q", secretName)
-	}
-
-	if key, ok = secret.Data["key"]; !ok {
-		return nil, fmt.Errorf("could not find TLS key in `key` field in secret %q", secretName)
-	}
-
-	cacert, hasCACert := secret.Data["cacert"]
-	insecureSkipVerifyRaw, hasInsecureSkipVerify := secret.Data["insecureSkipVerify"]
-
-	if !hasCACert && !hasInsecureSkipVerify {
-		return nil, fmt.Errorf("both cacert and insecureSkipVerify are not set. One of them need to be set to be able to call ETCD metrics")
-	}
-
-	// insecureSkipVerify is set to false by default, and can be overridden with the insecureSkipVerify field
-	insecureSkipVerify := false
-	if hasInsecureSkipVerify {
-		insecureSkipVerify = strings.ToLower(string(insecureSkipVerifyRaw)) == "true"
-	}
-
-	tlsConfig, err := parseTLSConfig(cert, key, cacert, insecureSkipVerify)
-	if err != nil {
-		return nil, fmt.Errorf("parsing TLS config: %w", err)
-	}
-
-	return tlsConfig, nil
-}
-
-func parseTLSConfig(certPEMBlock, keyPEMBlock, cacertPEMBlock []byte, insecureSkipVerify bool) (*tls.Config, error) {
```

Comment:
this was completely removed since k8s transport package handle this.

Reply:
🎉